pkg/pillar/cmd/upgradeconverter/upgradeconverterutils.go

```diff
@@ -0,0 +1,50 @@
+package upgradeconverter
+
+import (
+	"encoding/json"
+	log "github.com/sirupsen/logrus"
+	"io/ioutil"
+	"os"
+	"time"
+)
+
+// FilePathData - Stores old and new file paths
+type FilePathData struct {
+	OldVersionPath string
+	NewVersionPath string
+}
+
+func fileExists(filename string) bool {
+	info, err := os.Stat(filename)
+	if os.IsNotExist(err) {
+		return false
+	}
+	return !info.IsDir()
```

Comment:
This is broken because if the error is e.g., no permissions there is no point in looking at IsDir.

What semantics do you want? That the file exists, in which case it should return false if there is any error, and true otherwise. And log if the error is something other than IsNotExist